src/assets.zig

```diff
@@ -76,6 +122,20 @@ pub fn readAllZonFilesInAddons(externalAllocator: NeverFailingAllocator, addons:
 					std.log.err("Could not open {s}/{s}: {s}", .{subPath, entry.path, @errorName(err)});
 					continue;
 				};
+
+				// If this is migrations file, we interrupt normal asset processing and store it in migrations hashmap.
+				if(std.ascii.eqlIgnoreCase(entry.basename, "_migrations.zig.zon")) {
+					defer externalAllocator.free(id);
```

Comment:
If we only accept one file for the migrations, then it should be placed outside the loop, so we avoid any further accidental memory leaks of the stuff declared before it in the loop.
Basically it could just be
```
if(migrations != null) blk: {
    const zon = main.files.Dir.init(dir).readToZon(externalAllocator, "_migrations.zig.zon") catch |err| {
        if(err != error.FileNotFound) log error...
        break :blk;
    }
    migrations.?.put(externalAllocator.dupe(u8, addonName), zon);
}
```
(And of course you'd still need to ignore the file in the loop)